user-modules/gpg.nix

```diff
@@ -29,13 +24,12 @@
   services.gpg-agent = {
     enable = true;
     enableSshSupport = true;
-    pinentry.package =
-      if lib2.hasWorkload config "desktop" then pkgs.pinentry-qt else pkgs.pinentry-curses;
+    pinentry.package = pkgs.pinentry-qt;
```

Comment:
this needs a better check if there is a desktop enabled. set curses here and override in a desktop module